pkg/virt-controller/services/template.go

```diff
@@ -356,11 +356,37 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, t
 
 	var volumes []k8sv1.Volume
 	var volumeDevices []k8sv1.VolumeDevice
-	var userId int64 = 0
-	var privileged bool = false
 	var volumeMounts []k8sv1.VolumeMount
 	var imagePullSecrets []k8sv1.LocalObjectReference
 
+	var userId int64 = 0
+	var privileged bool = false
+	nonRoot := t.clusterConfig.NonRootEnabled()
```

Comment:
@xpivarc I think, in order to address the virtiofs incompatibility, we simply need to set this to non-root if `util.IsVMIVirtiofsEnabled` 
we could do the same for cpu pinning (if vmi has spec.domain.cpu.dedicatedCpuPlacement), until BZ#1935039 is resolved.